packages/http-client-csharp/emitter/src/lib/utils.ts

```diff
@@ -21,7 +21,7 @@ export async function execCSharpGenerator(
     "-g",
     options.generatorName,
   ];
-  if (options.newProject === true) {
+  if (options.newProject) {
```

Comment:
This needs to be a truthy check when using the CLI (unless we want to do the type coercion ourselves) as the CLI is not correctly parsing to boolean. Filed https://github.com/microsoft/typespec/issues/6597